experimental/lib/Transforms/LSQSizing/HandshakeSizeLSQs.cpp

```diff
@@ -233,8 +233,22 @@ std::optional<LSQSizingResult> HandshakeSizeLSQsPass::sizeLSQsForCFDFC(
     TimingDatabase timingDB, unsigned initialII, std::string collisions) {
 
   CFDFCGraph graph(funcOp, cfdfcBBs, timingDB, initialII);
-  auto loadOps = graph.getOperationsWithOpType<handshake::LSQLoadOp>();
-  auto storeOps = graph.getOperationsWithOpType<handshake::LSQStoreOp>();
+
+  // We only want LSQ loads, therefore we need to check if they are connected to
+  // an LSQ
+  auto allLoadOps = graph.getOperationsWithOpType<handshake::LoadOp>();
+  auto allStoreOps = graph.getOperationsWithOpType<handshake::StoreOp>();
+  std::vector<mlir::Operation *> loadOps;
+  std::vector<mlir::Operation *> storeOps;
+  for (auto op : allLoadOps) {
+    if (graph.isConnectedToLSQ(op))
+      loadOps.push_back(op);
+  }
+
+  for (auto op : allStoreOps) {
+    if (graph.isConnectedToLSQ(op))
+      storeOps.push_back(op);
+  }
```

Comment:
Maybe just do this instead (same thing for the stores)? You don't need `allLoadOps` later
```c++
std::vector<mlir::Operation *> loadOps;
for (auto op : graph.getOperationsWithOpType<handshake::LoadOp>) {
  if (graph.isConnectedToLSQ(op))
    loadOps.push_back(op);
}
```